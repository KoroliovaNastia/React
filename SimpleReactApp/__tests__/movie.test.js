import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Movie from '../src/components/movie';
import MovieDescription from '../src/components/movieDescription';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

configure({ adapter: new Adapter() });

const mockStore = configureStore();
const initialStore = {
    movieState: {
        movie: {id: '1', image: '../src/images/movie-default.jpg', title:'Not yet...', genres:['Adventure'], release_date: '2022', vote_average: 5, runtime: 154, description:""}
    }
}

const store = mockStore(initialStore);

describe('Movie component', () => {
    it('should be render correctly', () => {
        const component = shallow(<Movie.WrappedComponent movie={initialStore.movieState.movie} />)
        expect(shallowToJson(component)).toMatchSnapshot();
    })
})

describe('Movie description component', () => {
    it('should be render correctly', () => {
        const component = mount(<MovieDescription store={store} />)
        expect(shallowToJson(component)).toMatchSnapshot();
    })
})
