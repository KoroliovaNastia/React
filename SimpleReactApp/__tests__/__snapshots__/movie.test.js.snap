// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should be render correctly 1`] = `
<div
  className="movie"
>
  <Image />
  <Link
    to="/film/1"
  >
    Not yet...
  </Link>
  <ul>
    <p
      key="Not yet...Adventure"
    >
      Adventure
    </p>
  </ul>
  <p>
    2022
  </p>
  <p>
    5
  </p>
</div>
`;

exports[`Movie description component should be render correctly 1`] = `
<Connect(MovieDescription)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MovieDescription
    dispatch={[Function]}
    movie={
      Object {
        "description": "",
        "genres": Array [
          "Adventure",
        ],
        "id": "1",
        "image": "../src/images/movie-default.jpg",
        "release_date": "2022",
        "runtime": 154,
        "title": "Not yet...",
        "vote_average": 5,
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Image>
      <img />
    </Image>
    <div>
      <p>
        Not yet...
      </p>
      <p>
        5
      </p>
    </div>
    <div>
      <p>
        2022
         year 
        154
         min
      </p>
      <p />
    </div>
  </MovieDescription>
</Connect(MovieDescription)>
`;
